containers/flatpak/test/test-browser

```diff
@@ -11,8 +11,23 @@ TESTLIB = os.path.join(TEST_DIR, "test-lib.js")
 FLATPAK_ID = "org.cockpit_project.CockpitClient"
 JS_TIMEOUT = 15
 
+STORAGE = os.path.join(os.path.expanduser("~"), ".var/app/org.cockpit_project.CockpitClient/data/webkitgtk/localstorage/")
+TMP_STORAGE = "/tmp/localstorage"
+
 
 class BrowserTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        # Hide localstorage to start browser from clean state
+        subprocess.check_call(["mkdir", "-p", TMP_STORAGE])
+        subprocess.check_call(["cp", "-a", STORAGE, TMP_STORAGE])
+        subprocess.check_call(["rm", "-rf", STORAGE])
```

Comment:
Please avoid a hardcoded /tmp/localstorage path. That is likely to exist already from previous aborted runs (and then would wreak havoc), it changes time stamps and other attributes. Please just rename the whole existing `~/.var/app/org.cockpit_project.CockpitClient` directory to `*.cockpittest`, clean up the new one, and rename it back.

This will also fail if the directory does not exist at all -- I'm surprised that it doesn't crash and burn on the `cp` in the GitHub workflow test.

But this is still a nasty hack -- @allisonkarlitskaya , isn't there some way that we could use some `$XDG_*` variable here?

Reply:
I started an ubuntu-2004 standard bots VM and ran the steps in .github/workflows/flatpak-test.yml, but only invoked the browser test directly (not test-ssh):
```
ERROR: setUpClass (__main__.BrowserTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "containers/flatpak/test/test-browser", line 23, in setUpClass
    subprocess.check_call(["cp", "-a", STORAGE, TMP_STORAGE])
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cp', '-a', '/home/admin/.var/app/org.cockpit_project.CockpitClient/data/webkitgtk/localstorage/', '/tmp/localstorage']' returned non-zero exit status 1.
```

So it's test-ssh which avoids the crash. Doing a `mkdir -p` works around it.

Reply:
If you have the PID then you could mine the environment out of the running process using something like:

```
xargs -0n1 < /proc/172619/environ | grep XDG_DATA_HOME
```

Reply:
done in a fixup